### PR TITLE
test wheel from packaging pipeline with dockers

### PR DESCRIPTION
**Pull request by liqunfu**
It tests torch-ort with onnxruntime-training and pytorch stable (cuda 10.2 and 11.1) and nightly (cuda 10.2 and 11.1) builds. Upon success (one nightly), the nightly torch-ort wheel (only need one wheel for all configurations) is published to torch-ort channel.

New docker files adapted from the existing ones are added. Docker files are used in the tests.

### CODE REVIEW

**Review by asolms — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/packaging-pipeline.yml [hunk 1]
```diff
@@ -52,37 +75,54 @@ stages:
...
+            docker build -f $(PublicDockerFile) -t torchortpackagingtest .
+        displayName: "docker build -f $(PublicDockerFile) -t torchortpackagingtest"
+
+      # dockerfile has torch-ort. uninstall it to prepare for new install from current build
+      - task: CmdLine@2
+        inputs:
+          script: |
+            docker run --rm \
+              torchortpackagingtest \
+                /usr/bin/python3 -m pip uninstall -y torch-ort
```
Comment (suffiank) [inline tools/ci_build/github/azure-pipelines/packaging-pipeline.yml:91]: I'm confused. Won't the docker container be torn down after this uninstall command runs? That makes it useless, no?

**Review by suffiank — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/packaging-pipeline.yml (hunk 1 above)
Comment (liqunfu) [reply tools/ci_build/github/azure-pipelines/packaging-pipeline.yml:91]: good catch - removed --rm flag

Comment (suffiank) [reply tools/ci_build/github/azure-pipelines/packaging-pipeline.yml:91]: I don't think that's sufficient. The "--rm" flag just ensures the container doesn't leave a dead container around after the command completes. In the subsequent command, you'd have to re-start the dead container. Better yet, why not run the uninstall right before you plan to do the install Then both run in the same container.

Comment (liqunfu) [reply tools/ci_build/github/azure-pipelines/packaging-pipeline.yml:91]: Sorry @suffiank I misunderstood you and thanks also to @baijumeswani for telling me the same. I have removed that ineffective step. when torch-ort is installed it uses --upgrade to make sure the current wheel is used.

File: build.py [hunk 2]
```diff
@@ -20,16 +24,22 @@ def build_wheel(cwd, source_dir):
...
-    # installing torch-ort wheel
-    dist_path = os.path.join(source_dir, 'dist')
-    wheel_file = os.listdir(dist_path)[0]
-    run_subprocess([sys.executable, "-m", "pip", "install", "--upgrade", os.path.join(dist_path, wheel_file)], cwd)
+        # installing torch-ort wheel
+        dist_path = os.path.join(source_dir, 'dist')
+        wheel_file = os.listdir(dist_path)[0]
+        run_subprocess([sys.executable, "-m", "pip", "install", "--upgrade", os.path.join(dist_path, wheel_file)], cwd)
+    else:
+        print("cmd_line_args.wheel_file:", cmd_line_args.wheel_file)
```
[the remark this answers is not in the document]
Comment (liqunfu) [reply build.py:41]: Added:
print("With Devops pipeline, please confirm that the wheel file matches the one being built from a previous step.")

File: tools/python/upload_python_package_to_azure_storage.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        content_settings=content_settings)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Upload python whl to azure storage.")
+
+    parser.add_argument("--python_wheel_path", type=str, help="path to python wheel")
+    parser.add_argument("--account_name", type=str, help="account name")
+    parser.add_argument("--account_key", type=str, help="account key")
+    parser.add_argument("--container_name", type=str, help="container name")
```
[the remark this answers is not in the document]
Comment (liqunfu) [reply tools/python/upload_python_package_to_azure_storage.py]: added more descriptions

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**